### DISCUSSION TITLE
OpenOMF: official release for Windows for ARM?

### DISCUSSION THREAD

**Discussion by exocyt0sis** [omf2097/openomf#527] [Ideas]
More and more Windows users are on ARM processors. Would it be doable to add Windows for ARM as an official OpenOMF build target?

**Comment by katajakasa** [comment]
Technically possible. We just need someone to do it -- preferably someone with windows arm installation to test it with.

**Reply by exocyt0sis** [reply]
Thank you for your reply. While I have no real knowledge of building programs like these myself, I'm eager to help out with testing OpenOMF on Windows for ARM.

**Reply by katajakasa** [reply]
Ticket created at https://github.com/omf2097/openomf/issues/539 . We'll see if someone signs up to do the implementation ;)

**Reply by exocyt0sis** [reply]
> Ticket created at #539 . We'll see if someone signs up to do the implementation ;)

That sounds great! Meanwhile, I'll keep an eye on the project. If someone implements OpenOMF for Windows for ARM, please feel free to update this thread so everyone who follows it gets a notification. Thank you in advance! 👍

**Reply by talynone** [reply]
Looks like someone ported it here:

https://github.com/mijk84/win-arm64-binaries/releases/tag/openomf

**Reply by exocyt0sis** [reply]
> Looks like someone ported it here:
> 
> https://github.com/mijk84/win-arm64-binaries/releases/tag/openomf

Nice, however it's a two year old release that's also not a part of the official repository?

**Reply by talynone** [reply]
> > Looks like someone ported it here:
> > https://github.com/mijk84/win-arm64-binaries/releases/tag/openomf
> 
> Nice, however it's a two year old release that's also not a part of the official repository?

I just tested this, the source code links he has are misleading, it's change log for his release repository, not source code for the binary. Looks like it's version 0.6.6 and requires the "resources" folder that I got from the [windows download](https://github.com/omf2097/openomf/releases/tag/latest) zip file. 

Here's a screenshot of it running and shown in task manager as a ARM64 binary:

<img width="979" alt="image" src="https://github.com/user-attachments/assets/f364269f-05d0-48fd-9477-bd6b6a2a06b7">

**Reply by exocyt0sis** [reply]
Ah! While I haven't tested it out myself just yet, thank you for giving it ago. From what I understand, it belongs in the official repository, and not in an un-maintained offshoot so to speak. Here's hoping it'll end up in the main code branch!

**Reply by talynone** [reply]
> Ah! While I haven't tested it out myself just yet, thank you for giving it ago. From what I understand, it belongs in the official repository, and not in an un-maintained offshoot so to speak. Here's hoping it'll end up in the main code branch!

You'll have to message the author about that.

As a side note, the Vanilla DOS game runs pretty well through windows ARM 64 version of [DOSBOS-X](https://github.com/joncampbell123/dosbox-x/tree/master). 

You can even apply something like a xbrz scaler to make it look nicer, and DOSBOS-X supports IP over IPX networking so multiplayer should work too. 

<img width="1472" alt="image" src="https://github.com/user-attachments/assets/2d4d6225-06a9-4987-a263-c821d7e11b42">

**Reply by exocyt0sis** [reply]
> As a side note, the Vanilla DOS game runs pretty well through windows ARM 64 version of [DOSBOS-X](https://github.com/joncampbell123/dosbox-x/tree/master).
> 
> You can even apply something like a xbrz scaler to make it look nicer, and DOSBOS-X supports IP over IPX networking so multiplayer should work too.

Thank you for pointing it out. However, MS-DOS emulation isn't really what I'm looking for. I'd like to run OpenOMF _natively_ on ARM64 hardware running Windows. For that reason, I'm crossing my fingers official builds will turn up here in the near future.

**Reply by Nopey** [reply]
**EDIT: This PR has since been merged to master, see next comment.**

I've opened a draft PR https://github.com/omf2097/openomf/pull/781 that will allow Windows-on-ARM64 binaries to be built as part of CI.
I have no way of testing these binaries to find out if they run, but if you're able please let me know if you're able to get openomf_arm64.exe to run.

**Reply by Nopey** [reply]
Latest Windows releases now include an `openomf_arm64.exe` alongside the usual `openomf.exe`.
https://github.com/omf2097/openomf/releases/tag/latest

Happy robot-smashing, @exocyt0sis! and please do let us know if you encounter any bugs or issues with the Windows ARM port.
